session.go

```diff
@@ -807,14 +958,32 @@ func (s *session) handlePacket(p *receivedPacket) {
 	}
 }
 
+func (s *session) mitigationOn(encLevel protocol.EncryptionLevel) bool {
+	return s.config.AttackTimeout > 0 && !s.handshakeComplete && encLevel != protocol.Encryption1RTT
+}
+
+// MaybeRecover runs fail(err) immediately if injection attack mitigation is turned off,
+// or after AttackTimeout elapses if no packets are received
+func (s *session) maybeRecover(fail func(error), err error) {
```

Comment:
I think `fail` here and throughout should probably be called `shutdown` or similar, since it's not necessarily a failure at all, but normally a deliberate remote close.